internal/backend/kubernetes/backend.go

```diff
@@ -2,104 +2,80 @@ package kubernetes
 // between the cluster and internal caches. Consumers can wait for the initial sync using WaitForCachesync().
 // See k8s.io/Backend-go/tools/Backendcmd for constructing *rest.Config objects.
-func New(config Config) (*Backend, error) {
+func NewBackend(cfg Config) (*Backend, error) {
 	opts := tinkcontrollers.GetServerOptions()
-	opts.Namespace = config.Namespace
+	opts.Namespace = cfg.Namespace
 
 	// Use a manager from the tink project so we can take advantage of the indexes and caching it configures.
 	// Once started, we don't really need any of the manager capabilities hence we don't store it in the
 	// Backend
-	manager, err := tinkcontrollers.NewManager(config.Config, opts)
+	manager, err := tinkcontrollers.NewManager(cfg.Config, opts)
 	if err != nil {
 		return nil, err
 	}
 
-	managerCtx, cancel := context.WithCancel(context.Background())
+	// Default the context.
+	if cfg.Context == nil {
+		cfg.Context = context.Background()
+	}
+
 	go func() {
-		if err := manager.Start(managerCtx); err != nil {
+		if err := manager.Start(cfg.Context); err != nil {
 			panic(err)
 		}
 	}()
 
-	backend := NewBackendWithClient(manager.GetClient())
-	backend.close = cancel
-	backend.waitForCacheSync = manager.GetCache().WaitForCacheSync
-
-	return backend, nil
-}
-
-// ListerClient lists Kubernetes resources using a sigs.k8s.io/controller-runtime Backend.
-type ListerClient interface {
-	List(ctx context.Context, list crclient.ObjectList, opts ...crclient.ListOption) error
-}
-
-// NewBackendWithClient creates a new Backend instance that uses Backend to find resources.
-// The Close() and WaitForCacheSync() methods of the returned Backend are noops.
-func NewBackendWithClient(client ListerClient) *Backend {
-	return &Backend{
-		client:           client,
-		close:            func() {},
-		closeM:           &sync.RWMutex{},
-		waitForCacheSync: func(context.Context) bool { return true },
+	backend := &Backend{
+		client:           manager.GetClient(),
+		closer:           cfg.Context.Done(),
+		WaitForCacheSync: manager.GetCache().WaitForCacheSync,
 	}
-}
 
-// Close stops synchronization with the cluster. Subsequent calls to all of k's methods, excluding IsHealthy(), have
-// undefined behavior.
-func (b *Backend) Close() {
-	b.closeM.Lock()
-	defer b.closeM.Unlock()
-
-	// nil indicates Close() has been called for other methods.
-	b.close()
-	b.close = nil
-}
-
-// WaitForCacheSync waits for the internal Backend cache to synchronize.
-func (b *Backend) WaitForCacheSync(ctx context.Context) bool {
-	return b.waitForCacheSync(ctx)
+	return backend, nil
 }
 
-// IsHealthy returns true until k.Close() is called. It is thread safe.
+// IsHealthy returns true until the context used to create the Backend is cancelled.
 func (b *Backend) IsHealthy(context.Context) bool {
-	b.closeM.RLock()
-	defer b.closeM.RUnlock()
-	return b.close != nil
+	select {
+	case <-b.closer:
```

Comment:
whats the difference between b.closer and the context passed in? why not just use that one?

Reply:
This is from the context, we're just not holding a reference to the whole context because storing a context felt dirty in itself.